src/client/components/AllLogs.vue

```diff
@@ -173,13 +196,41 @@ export default {
       console.log(`Deleting log "${payload.name}"...`);
     },
     syncAll() {
+      // updateAllLogs sends un-synced logs to the server.
       function logSyncer(log) {
         return {
           ...log,
           isReadyToSync: true,
         };
       }
       this.$store.commit('updateAllLogs', logSyncer);
+      this.readyToGetLogs = true;
+      // ReadyToGetLogs will be set back to false when this.logs updates, meaning the send is complete
+    },
+    // This function is only used for testing with the log search box
+    getLogs() {
+      if (this.logSearchTerms === ''){
+        this.$store.dispatch('getLogs', '');
+      } else if (isNaN(this.logSearchTerms)) {
+        this.$store.dispatch('getLogs', JSON.parse('{'+this.logSearchTerms+'}'));
+      } else {
+        this.$store.dispatch('getLogs', Number(this.logSearchTerms));
+      }
+    },
+  },
+  watch: {
```

Comment:
I'm not sure why this watcher is necessary. I'm also a little confused by the role of the `readyToGetLogs` state. @alexadamsmith , can you explain this a little?

Reply:
This will be handled instead by wrapping sendLogs in a promise, as we discussed.